releasenotes/notes/handle-pipeline-crashes-and-resume-state-21c0e46a8935cbe2.yaml

```diff
@@ -0,0 +1,6 @@
+---
+features:
+  - |
+    If an error occurs during the execution of a pipeline, the pipeline will raise an exception containing, besides the
+    error message, the components and it's inputs up to the point of failure. This allows you to inspect and debug the
+    the pipeline up to the point of failure.
```

Comment:
In this case I believe we will just be returning the outputs of the pipeline at this point. 

Let's also mention that we will be raising the `PipelineRuntimeError`